src/context/AppProvider/hooks/useDiscoverAccounts.tsx

```diff
@@ -0,0 +1,150 @@
+          const walletId = await wallet.getDeviceID()
+          const isMultiAccountWallet = wallet.supportsBip44Accounts()
+          const currentPortfolio = portfolio.selectors.selectPortfolio(store.getState())
+
+          let accountNumber = 0
+          let hasActivity = true
+          let isDegraded = false
+          const chainAccountMetadata: AccountMetadataById = {}
+
+          while (hasActivity) {
+            if (accountNumber > 0) {
+              if (!isMultiAccountWallet) break
+              if (isMetaMaskMultichainWallet && !isSnapInstalled) break
+            }
+
+            try {
+              const accountIdWithActivityAndMetadata = await getAccountIdsWithActivityAndMetadata(
+                accountNumber,
+                chainId,
+                wallet,
+                Boolean(isSnapInstalled),
+              )
+
+              hasActivity = accountIdWithActivityAndMetadata.some(account => account.hasActivity)
+
+              if (hasActivity || accountNumber === 0) {
+                accountIdWithActivityAndMetadata.forEach(({ accountId, accountMetadata }) => {
+                  chainAccountMetadata[accountId] = accountMetadata
+                })
+              }
+
+              accountNumber++
+            } catch (error) {
+              console.error(`Error discovering accounts for chain ${chainId}:`, error)
+              isDegraded = true
+              break
+            }
+          }
+
+          if (Object.keys(chainAccountMetadata).length > 0) {
+            dispatch(
+              portfolio.actions.upsertAccountMetadata({
+                accountMetadataByAccountId: chainAccountMetadata,
+                walletId,
+              }),
+            )
+
+            Object.keys(chainAccountMetadata).forEach(accountId => {
+              // Don't enable accounts that are already in the portfolio so we keep it disabled if user manually disabled it
+              if (
+                currentPortfolio.accountMetadata.byId[accountId] &&
+                currentPortfolio.wallet.byId[walletId]?.includes(accountId)
+              )
+                return
+
+              dispatch(portfolio.actions.enableAccountId(accountId))
+            })
+          }
+
+          return {
+            accountMetadataByAccountId: chainAccountMetadata,
+            hasActivity,
+            isDegraded,
+            chainId,
+          }
+        },
+        staleTime: Infinity,
+        gcTime: Infinity,
+        enabled: Boolean(wallet && deviceId),
+      })),
+    [dispatch, isSnapInstalled, wallet, deviceId, supportedChainIds, connectedRdns],
+  )
+
+  const accountsDiscoveryQueries = useQueries({
+    queries,
+  })
+
+  const { isLoading, isFetching } = useMemo(() => {
+    return {
+      isLoading: accountsDiscoveryQueries.some(query => query.isLoading),
+      isFetching: accountsDiscoveryQueries.some(query => query.isFetching),
+    }
+  }, [accountsDiscoveryQueries])
+
+  const degradedChainIds = useMemo(() => {
```

Comment:
suggestion - `isSome` filter so we don't have to deal with `undefined`?

Reply:
Why not but I don't really get what will be the benefits as we are spreading to merge both heavy `getAccounts` rtk and the account discovery afterwhile

Reply:
Closing to avoid getting this stale, if we want to change something, don't hesitate to ping me for follow ups